_posts/2020-03-19-orgwide-ecr.md

```diff
@@ -0,0 +1,108 @@
+our Elastic Container Registries (ECR).
+The Core Platform team has ECRs to host Docker containers which can and
+should be consumed by other teams and resources in their AWS Accounts. Not
+only that, we also need to access our own containers from different accounts.
+As a matter of habit, anything "production", we deploy in our "production"
+Account, with strong access control policies and security, such as read-only
+access to the AWS Console. We do our normal development and iteration in a
+"development" Account, which may be host to any number of AWS Elastic Container
+Services (ECS), each needing to pull containers from those ECRs.
+
+Even within a single team, we're using multiple AWS Accounts, and have
+cross-account IAM policies to implement!
+
+I recently watched a demo in a team meeting from my colleague
+[QP](https://github.com/houqp) who was setting up IAM cross-account Roles.
+Based on his demo, I knew that getting the cross-account Roles correct for our
+ECR use-cases was going to be tedious and painful. I lamented this to our
+friends at [The Duckbill Group](https://www.duckbillgroup.com/), as I usually
+do whenever something in AWS feels unpleasant. "Surely I'm missing something
+here." Luckily enough, I was missing something:
+
+```
+ rtyler | I'm assuming there's no feature I'm missing which would allow us to say "any resource in our AWS org can access this", I
+        | kind of really want a global read-only access for some ECRs :/
+ rtyler | is there an arn shortcut for "whole org" perhaps?
+cquinn* | Yes, the AWS:PrincipalOrgID Condition Key.
+cquinn* | https://aws.amazon.com/blogs/security/control-access-to-aws-resources-by-using-the-aws-organization-of-iam-principals/
+        | goes into some depth.
+ rtyler | oh god, turing complete JSON
+cquinn* | Cheer up, I’m sure it works in YAML.
+```
+
+Thankfully, [Corey](twitter.com/QuinnyPig/) was 100% correct, the
+`AWS:PrincipalOrgID` condition in the IAM policy document would allow the exact
+type of quasi-global read-only access I was after. Below is a snippet of
+Terraform which defines the policy:
+
+```terraform
+data "aws_iam_policy_document" "ecr_readonly_access" {
+  statement {
+    sid    = "ReadonlyAccess"
+    effect = "Allow"
+
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+
+    condition {
+      test     = "StringLike"
+      variable = "aws:PrincipalOrgID"
+      # This is our organization-wide identifier which can be found after
+      # log-in to AWS: <https://console.aws.amazon.com/organizations/home>
+      values = ["o-REDACTED"]
+    }
+
+    actions = [
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:GetRepositoryPolicy",
+      "ecr:DescribeRepositories",
+      "ecr:ListImages",
+      "ecr:DescribeImages",
+      "ecr:BatchGetImage",
+      "ecr:DescribeImageScanFindings",
+    ]
+  }
+}
```

Comment:
would be useful to also include `aws_ecr_repository_policy` resource here to demonstrate how to attach this policy to a repo.